### PR TITLE
feat: injest ethEventsProvider event logs in batches

### LINKED ISSUE #317: feat: EthEventsProvider should injest event logs in batches.
**What is the feature you would like to implement?**
When the `EthEventsProvider` (https://github.com/farcasterxyz/hub/blob/main/src/storage/engine/flatbuffers/providers/ethEventsProvider.ts) is first created, it connects to an Ethereum RPC node and injests historical logs. 

Right now, it just gets all logs since the start in one go, but it should probably be fetched in batches of ~10k blocks.

**Why is this feature important?**
As more blocks pass, the chances of timeouts and other errors increases. We should batch the incoming logs to make them easier to be processed. 

**How should this feature be built? (optional)**
* Add a way to batch the calls to `getHistoricalIdEvents` (https://github.com/farcasterxyz/hub/blob/main/src/storage/engine/flatbuffers/providers/ethEventsProvider.ts#L196)
* Add a way to batch the calls to `getHistoricalNameEvents` (https://github.com/farcasterxyz/hub/blob/main/src/storage/engine/flatbuffers/providers/ethEventsProvider.ts#L229)

**Additional context**
Bonus - Add retry logic for failed blocks

### PR DESCRIPTION
**Pull request by TheLDB**
## Motivation
This PR resolves issue #317, related to injesting the event logs from smart contracts in batches, vs all at once.

tl;dr for #317 is that trying to injest hundreds of thousands of events in one go, can lead to timeout and associated bugs, but processing them in smaller batches will allow them to be processed easier

## Change Summary
I updated the `syncHistoricalIdEvents` and `syncHistoricalNameEvents` functions to use a configurable batchSize parameter. This allows event logs to be injested in batches, vs all at once.

Along with that, although its outside of the scope of the issue, I added a section for Editor cache to the .gitignore, including .vscode, .idea, and .fleet, since I didn't want to accidentally commit my cache

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [x] The title of this PR adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [x] The PR has been tagged with the appropriate change type label(s) (i.e. documentation, feature, bugfix, or chore)
- [] Changes to the protocol specification have been merged - N/A, I don't think theres any protocol spec that needs to be updated/merged for this PR? Happy to update it if there is, though.

## Additional Context
N/A

### CODE REVIEW

**Review by neverabsolute — APPROVED** [review]
lgtm 👍

**Comment by imfunniee** [comment]
landon is 10xdev i approve it

**Comment by jacc** [comment]
vouch

**Comment by adityapk00** [comment]
Looks good to me! Please go ahead and merge this!

**Review by adityapk00 — APPROVED** [review]

**Comment by TheLDB** [comment]
@adityapk00 not able to merge btw, gh says "Only those with write access to this repository can merge pull requests."

**Comment by adityapk00** [comment]
Done, merged it on your behalf :)
